src/main/java/com/example/provider/VehicleServlet.java

```diff
@@ -82,35 +88,61 @@ public VehicleServlet(
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     ServletUtils.setStandardResponseHeaders(response);
-
+    
     String vehicleId;
 
+    // Check if there is a vehicle_id
     try {
       vehicleId = ServletUtils.getEntityIdFromRequestPath(request);
-    } catch (IllegalArgumentException e) {
-      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "No Vehicle id provided.");
-      return;
+    }
+    catch (IllegalArgumentException e) {
+        response.sendError(HttpServletResponse.SC_BAD_REQUEST, "No Vehicle id provided.");
```

Comment:
I think this exception will be thrown if no vehicle id is provided so your changes below won't be reached.